vllm/model_executor/models/utils.py

```diff
@@ -0,0 +1,41 @@
+import torch
+
+from vllm.multimodal import BatchedTensors
+
+
+def merge_vision_embeddings(input_ids: torch.Tensor,
+                            inputs_embeds: torch.Tensor,
+                            vision_embeddings: BatchedTensors,
+                            image_token_id: int) -> torch.Tensor:
```

Comment:
Shouldn't we have this function inside `multimodal/utils` since this is only relevant to VLMs?

Personally I'm also not a huge fan of making this a util function - IMO merging vision embeddings & input embeddings is part of the model implementation (it just happened that all the VLMs share the same behavior now), so keeping all relevant functions in the same place is better for readability.

Reply:
That's why I placed this function under models directory rather than multimodal directory since it's specific to models only. Do you want me to copy this to each model file individually?

Reply:
I think having some repetitive code in this case is okay since this is only relevant to those specific VLMs.

My only concern of keeping it this way is that we don't know if this is generalizable to all future VLMs, so in the future we might have some models that use this shared util function and other models specify their individual embedding processing function inside their model.py file, and thus it'll might be confusing to the developers whether they should keep all the functions used in the model implementation within the same file, or specify it in another location.

Either way, this is just my nit and you don't have to necessarily agree with it.

Reply:
I think we probably have to redesign the framework anyway if we need other ways of merging the embeddings. @xwjiang2010 since you came up with this method in the first place, what are your thoughts on this?